xbmc/PlayListPlayer.cpp

```diff
@@ -946,7 +947,11 @@ void PLAYLIST::CPlayListPlayer::OnApplicationMessage(KODI::MESSAGING::ThreadMess
     {
       // Discard the current playlist, if TMSG_MEDIA_PLAY gets posted with just a single item.
       // Otherwise items may fail to play, when started while a playlist is playing.
-      Reset();
+      // But a single item in a stack is allowed.
+      CPlayList& playlist = GetPlaylist(m_iCurrentPlayList);
+      CFileItemPtr pitem = playlist[m_iCurrentSong];
```

Comment:
We want to get rid of `typedef`s like `CFileItemPtr`. Please use the real type instead. Here, and everywhere else in this PR.


`const std::shared_ptr<CFileItem> pitem...`